### ISSUE TITLE
workaround: incorrect top label predictions when using lime_image with pycaffe

### ISSUE THREAD

**Issue by maxkohlbrenner** [marcotcr/lime#92]
Hi,

I am currently using lime_image as in https://marcotcr.github.io/lime/tutorials/Tutorial%20-%20images.html but using pycaffe instead of tensorflow for the neural network. 
In this case, the top label predictions returned by lime_image.data_label are incorrect if the classifier function gets called several times. 
This is probably due to the fact that my classification function directly returns the out['prob'] values of the caffe blob which is not a deep copy. 
For me, changing lines 239 and 243 in lime_image.py from 

labels.extend(preds)

to 

labels.extend(copy.deepcopy(preds))

resolved the issue.
Just wanted to let you know about this since I guess it might be a general problem for certain classification functions passed to lime_image. 

Best regards,
Max

**Comment by marcotcr** [comment]
Thanks! I think this is rare enough not to warrant a change
